text/####-error-mitigation.md

```diff
@@ -0,0 +1,235 @@
+```
+def available_qubits(self, stretch_factors: List) -> List:
+    """Returns a list of qubits for which the given stretch factors are available."""
+```
+The user may also query the backend to find out how the error mitigation is configured.
+```
+def richardson_error_mitigation_config(self) -> List:
+    """Returns a List of dict showing the available configuration of the richardson error mitigation."""
+```
+The returned list would contain dictionaries of the form
+```
+{'stretch_factors': list, 'qubits': list}.
+```
+For example, the list
+```
+[
+  {'stretch_factors': ['1.0', '1.1', '1.25'], 'qubits': [0, 1, 2, 3]},
+  {'stretch_factors': ['1.0', '1.1'], 'qubits': [0, 1, 2, 3, 4]}
+]
+```
+would imply that qubit 4 does not have stretch factor 1.25 available.
+
+To execute a quantum circuit with error mitigation, the user would do
+```
+execute(circ, backend, ..., error_mitigation='richardson')
+```
+to use all the stretch factors available to the qubits in the quantum register of `circ`.
+Additionally, the user may specify a subset of stretch factors to use by doing
+```
+execute(circ, backend, ..., error_mitigation='richardson', stretch_factors=[1.0, 1.25, 1.5]).
+```
+This will require a validation step, done in Qiskit Terra and based on the `config`, to check that the specified stretch factors are available for all of the qubits in the quantum register of `circ`.
+If this is not the case, an error will be raised.
+Here, `error_mitigation` specifies the error mitigation method to use.
+This value would by default be `None` when error mitigation is not used.
+The changes needed in Qiskit to implement error mitigation would require changes in the `assemble` function.
+`assemble` would detect that the user requires error mitigation and replace each circuit by several circuits, one for each stretch factor.
+The `experiments` in the resulting `qobj` would have an entry in the header that specifies the stretch factor.
+For example, the `qobj` for two circuits (`circA` and `circB`) to be run with error mitigation with three stretch factors would look like
+```
+print(qobj.experiments[0])
+{
+    'config': {...},
+    'header': {'name': 'circA_ef01a', 'stretch_factor': '1.0', ...},
+    'instructions': {...}
+}
+
+print(qobj.experiments[1])
+{
+    'config': {...},
+    'header': {'name': 'circA_ef01b', 'stretch_factor': '1.1', ...},
+    'instructions': {...}
+}
+print(qobj.experiments[3])
+{
+    'config': {...},
+    'header': {'name': 'circB_ec01a', 'stretch_factor': '1.0', ...},
+    'instructions': {...}
+}
+
+print(qobj.experiments[4])
+{
+    'config': {...},
+    'header': {'name': 'circB_ec01b', 'stretch_factor': '1.1', ...},
+    'instructions': {...}
+}
+```
+Here, the instructions for experiments 0, 1, and 2 are identical since they correspond to `circA`.
+The instructions for experiments 3, 4, and 5, are also identical since they correspond to `circB`.
```

Comment:
Seems like the above code snippet should include 5 experiment examples.

Reply:
Why 5? Two circuits and three stretch factors results in 6 experiments. Of the six experiments, four are displayed. I'll add a bit more to the text to explain this more.

Reply:
yes 6 is correct